Index: sql/sql_acl.cc
===================================================================
--- sql/sql_acl.cc	(revision 17036)
+++ sql/sql_acl.cc	(revision 17035)
@@ -9486,11 +9486,6 @@
     mysql_mutex_unlock(&LOCK_connection_count);
     if (!count_ok)
     {                                         // too many connections
-      if (thd->user_connect)
-      {
-        decrease_user_connections(thd->user_connect);
-        thd->user_connect= 0;
-      }
       my_error(ER_CON_COUNT_ERROR, MYF(0));
       DBUG_RETURN(1);
     }
